### ISSUE TITLE
Allow & immediately after element name

### ISSUE THREAD

**Issue by subhaze** [sass/sass#1369]
div& causes an error

```
Syntax error: Invalid CSS after "   div": expected "{", was "& {"

"&" may only be used at the beginning of a compound selector.
```

Expecting

``` scss
.some-class {
  div& {
    ...
  }
}
```

to become

``` css
div.some-class {
  ...
}
```

**Comment by hail2u** [comment]
You can do this with `@at-root` and `#{&}`:

``` scss
.some-class {
  @at-root div#{&} {
    ....
  }
}
```

**Comment by subhaze** [comment]
@hail2u thanks for the heads-up on that. I'm moving over from LESS due to work so I'm still trying to get my bearings in SCSS.

I'm curious of the reason for this extra syntax since the docs state `&` as a reference to the parent selector.

**Comment by chriseppstein** [comment]
As mentioned by @hail2u, This is addressed in 3.4
